app/connector/mongodb/src/main/java/io/syndesis/connector/mongo/MongoClientCustomizer.java

```diff
@@ -15,36 +15,46 @@
  */
 package io.syndesis.connector.mongo;
 
-import java.util.Map;
-
+import com.mongodb.MongoClient;
+import com.mongodb.MongoClientURI;
+import io.syndesis.integration.component.proxy.ComponentProxyComponent;
+import io.syndesis.integration.component.proxy.ComponentProxyCustomizer;
 import org.apache.camel.CamelContext;
 import org.apache.camel.CamelContextAware;
+import org.apache.camel.Exchange;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import com.mongodb.MongoClient;
-import com.mongodb.MongoClientURI;
-
-import io.syndesis.integration.component.proxy.ComponentProxyComponent;
-import io.syndesis.integration.component.proxy.ComponentProxyCustomizer;
+import java.util.Arrays;
+import java.util.Map;
 
 public class MongoClientCustomizer implements ComponentProxyCustomizer, CamelContextAware {
     private static final Logger LOGGER = LoggerFactory.getLogger(MongoClientCustomizer.class);
 
+    private String operation;
     private CamelContext camelContext;
 
     @Override
-    public void setCamelContext(CamelContext camelContext) {
-        this.camelContext = camelContext;
+    public CamelContext getCamelContext() {
+        return this.camelContext;
     }
 
     @Override
-    public CamelContext getCamelContext() {
-        return this.camelContext;
+    public void setCamelContext(CamelContext camelContext) {
+        this.camelContext = camelContext;
     }
 
     @Override
     public void customize(ComponentProxyComponent component, Map<String, Object> options) {
+        // Validate operation parameter, set only once
+        if (operation == null) {
+            Object operation = options.get("operation");
+            if (operation instanceof String) {
+                this.operation = (String) operation;
+            }
+            component.setBeforeProducer(this::checkValidOperation);
```

Comment:
Why do we have to check the operation using a before producer? Can't we just validate the operation in the customize and throw exception right away. This way we would not need to save the operation to a local member variable in this class.

Reply:
I tried to find a way to know if the `customize` method is called by a producer or a consumer, since the check must be performed only by a producer endpoint. Did not find a way to identify that, any suggestion? an alternative would be to create a further customizer and configure for producer only, but I'd like to listen any alternative approach.

Reply:
I finally created a dedicated customizer just to check the operation.